apps/portal/app/buckets/[address]/_components/object.tsx

```diff
@@ -122,70 +128,59 @@ export default function Object({
+                `${objectSize.val} ${objectSize.unit || ""}`}
+            </span>
+            <span>
+              Expires {objectBlockDiff && objectBlockDiff < 0n ? "d" : ""}{" "}
+              {objectExpiryDisplay}
+            </span>
+          </div>
+          <div className="flex items-center gap-3">
             {deletePending || deleteReceiptFetching ? (
-              <Loader2 className="ml-auto animate-spin" />
+              <Loader2 className="animate-spin" />
             ) : (
-              <Trash
-                className="hover:text-destructive ml-auto opacity-20 hover:cursor-pointer hover:opacity-100"
-                onClick={handleDelete}
-              />
+              <div title="Delete object">
+                <Trash
+                  className="hover:text-destructive opacity-20 hover:cursor-pointer hover:opacity-100"
+                  onClick={handleDelete}
+                />
+              </div>
             )}
             <Link
               href={`${objectApiUrl}/v1/objects/${bucketAddress}/${encodeURIComponent(path)}`}
               target="_blank"
               className="opacity-20 hover:cursor-pointer hover:opacity-100"
+              title="Download object"
             >
               <Download />
             </Link>
-          </CardTitle>
-        </CardHeader>
-        <CardContent className="grid grid-cols-1 gap-14 sm:grid-cols-2">
-          <Metric
-            title="Blob Hash"
-            value={
-              <CollapsedStringDisplay
-                value={object.blobHash}
-                showCopy
-                copyTooltip="Copy blob hash"
-                copySuccessMessage="Blob hash copied"
-              />
-            }
-            valueTooltip={object.blobHash}
-          />
-          <Metric
-            title="Recovery Hash"
-            value={
-              <CollapsedStringDisplay
-                value={object.recoveryHash}
-                showCopy
-                copyTooltip="Copy recovery hash"
-                copySuccessMessage="Recovery hash copied"
-              />
-            }
-            valueTooltip={object.recoveryHash}
-          />
-          <Metric
-            title="Size"
-            value={objectSize.val}
-            subtitle={objectSize.unit}
-          />
-          <Metric
-            title={`Expire${(objectBlockDiff || 1) < 0 ? "d" : "s"}`}
-            value={objectExpiryDisplay}
-            valueTooltip={objectExpiryIso}
-          />
-          <div className="flex flex-col gap-2 sm:col-span-2">
-            <span className="text-muted-foreground text-xs">Metadata</span>
-            <pre className="text-muted-foreground min-h-12 border p-4 font-mono">
-              {arrayToDisplay(object.metadata)}
-            </pre>
+            <ShareButton
+              url={getPortalShareUrl()}
+              tooltip="Copy portal link to share"
+              successMessage="Portal link copied to clipboard"
+            />
```

Comment:
The icon in this button is small compared to the trash and download icons next to it. Will look into that component and leave more comments there.